tests/unautomated/test_laser_envelope.py

```diff
@@ -0,0 +1,396 @@
+# Copyright 2016, FBPIC contributors
+# Authors: Remi Lehe, Manuel Kirchen
+# License: 3-Clause-BSD-LBNL
+"""
+This test file is part of FB-PIC (Fourier-Bessel Particle-In-Cell).
+
+It tests the structures implemented in fields.py,
+by studying the propagation of a laser in vacuum:
+- The mode 0 is tested by using a linearly polarized gaussian beam,
+   which propagates to the right.
+- The mode 1 is tested by ????
+
+In both cases, the evolution of the a0 and w0 are compared
+with theoretical results from diffraction theory.
+
+These tests are performed in 3 cases:
+- Periodic box : in this case, a very large timestep is taken,
+  so as to test the absence of a Courant condition
+- Moving window : in this case, the timestep has to be reduced,
+  in order to leave time for the moving window to shift/damp the fields
+- In a galilean frame : in this case again, large timestep can be taken
+
+WARNING: Only Periodic box for mode 0 test is implemented for now
+
+Usage :
+-------
+In order to show the images of the laser, and manually check the
+agreement between the simulation and the theory:
+$ python tests/test_laser.py
+(except when setting show to False in the parameters below)
+
+"""
+import numpy as np
+from scipy.constants import c, m_e, e
+from scipy.optimize import curve_fit
+from fbpic.main import Simulation
+from fbpic.lpa_utils.laser import add_laser_pulse, \
+    GaussianLaser, LaguerreGaussLaser
+
+# Parameters
+# ----------
+# (See the documentation of the function propagate_pulse
+# below for their definition)
+show = False # Whether to show the plots, and check them manually
+
+use_cuda = False
+
+# Simulation box
+Nz = 400
```

Comment:
Actually, the envelope solver should work also when the laser wavelength is badly resolved ; therefore, I am wondering if using a small `Nz` (e.g. `Nz = 80`) would still work for this test. 
If yes, it would be great to modify this parameter accordingly.